tests/Simplify/PlusPlusTest.elm

```diff
@@ -45,13 +46,34 @@ a = "a" ++ ""
 a = "a"
 """
                         ]
-        , test """should not report x ++ "" (because this can lead to better performance)""" <|
+        , test """should not report x ++ "" when doNotSimplifyMicroOptimizedCode is enabled (because this can lead to better performance)""" <|
```

Comment:
In practice, we could fine-tune this kind of situation, even with micro-optimizations allowed.

```elm
x ++ "" -- report, it's useless
x ++ y ++ "" -- Don't report, it's useful
"foo" ++ y ++ "" -- report, it's useless
```